src/runtime/execution.rs

```diff
@@ -568,6 +570,35 @@ impl ExecutionState {
             self.current_span_entered = Some(unsafe { extend_span_entered_lt(self.current_span.enter()) });
         }
     }
+
+    fn set_tag_internal(&mut self, tag: u64) {
+        self.current_mut().set_tag(tag);
+    }
+
+    pub(crate) fn set_tag(tag: u64) {
+        ExecutionState::with(|s| {
+            s.set_tag_internal(tag);
+        });
+    }
+
+    fn get_tag_internal(&self) -> u64 {
+        match self.try_current() {
+            Some(current) => current.get_tag(),
+            None => 0,
+        }
+    }
+
+    pub(crate) fn get_tag() -> u64 {
+        ExecutionState::with(|s| s.get_tag_internal())
+    }
```

Comment:
I think that `get_tag` shouldn't call `get_tag_internal`. The default tag `0` should only be used to initialize the "root task", but not returned if `get_tag` is called in a wrong context.

Reply:
I dunno. Again, wrong context is really not possible in user land. This is as it is so that setup (line 288: `let tag = state.get_tag_internal();`) can be done in an easy/readable way.

Reply:
To make it really readable the function name could be `get_tag_or_default` or something similar. (and maybe make `0` a constant with a descriptive name)